### ISSUE TITLE
Cryptomator does not load the native appindicator libs

### ISSUE THREAD

**Issue by purejava** [flathub/org.cryptomator.Cryptomator#50]
```java
19:10:15.412 [JavaFX Application Thread] INFO  o.p.appindicator.RuntimeHelper - Native code library libayatana-appindicator3.so.1 failed to load
19:10:15.412 [JavaFX Application Thread] INFO  o.p.appindicator.RuntimeHelper - Native code library libappindicator3.so.1 failed to load
```

This is due to the appindicator Java bindings trying to load the native appindicator libs from ld.so.conf file paths, see https://github.com/purejava/appindicator-gtk3-java/blob/e283abed6835029406aef444c9220540480c49d0/src/main/java/org/purejava/appindicator/RuntimeHelper.java#L40C1-L74C6.

This is fine for native apps, but does not work for flatpak, as e.g. `/usr/lib` is hidden from the sandbox.
As flatpaked Cryptomator compiles its own libappindicator anyway and installs it to `/app/lib`, the proper solution is to add `/app/lib` to the libraries search path.

**Comment by infeo** [comment]
As a workaround for now, we can symlink the library, right? e.g. create in the flatpak `/usr/lib` pointing to `/app/lib`.

**Comment by infeo** [comment]
Ok, nevermind, this is not possible, The runtime reserves already `/usr/`. I need to investigate more.

**Comment by purejava** [comment]
> As a workaround for now, we can symlink the library, right? e.g. create in the flatpak `/usr/lib` pointing to `/app/lib`.

Unfortunately, this wouldn't be enough to fix this issue. Commonly, the native lib used is named `libappindicator3.so.1` on Linux, and that is what the Java bindings [look for](https://github.com/purejava/appindicator-gtk3-java/blob/e283abed6835029406aef444c9220540480c49d0/src/main/java/org/purejava/appindicator/RuntimeHelper.java#L32) and try to load.

However, the libappindicator lib, compiled by the flatpaked Cryptomator and put to `/app/lib`, is named `libappindicator3.so`, therefore would not be recognized or loaded by the Java bindings. So we need a new version of the Java bindings, that are compatible to flatpak. I am working on a new release of the Java bindings anyway, as I am making them compile with Java 21 and plan to release it within the next couple of days, including a fix for this issue.

**Comment by infeo** [comment]
I noticed, that the property `java.library.path` is not used. Couldn't a third/last-ditch attempt be implemented, such that
1. Check `LD_CONFIG`
2. Check `/usr/lib`
3. Check all paths in system property [`java.library.path`](https://docs.oracle.com/en/java/javase/20/docs/api/java.base/java/lang/System.html#java.library.path)

Or add system properties to allow setting the target paths explicitly? Just some ideas.

**Comment by purejava** [comment]
This issue should be fixed with [appindicator-gtk3-java release 1.3.4.](https://github.com/purejava/appindicator-gtk3-java/releases/tag/1.3.4).

I'll need to build `integrations-linux`, containing the new release, locally and afterwards build the flatpaked Cryptomator to test the changes.

**Comment by purejava** [comment]
TWIMC I straced the system-bus to get an idea, what is causing the warning.

Unfortunately, there is nothing happening on the system-bus, that is related, according to the timestamp (Cryptomator is on CEST):
```
** (cryptomator:2): WARNING **: 10:05:36.431: atk-bridge: get_device_events_reply: unknown signature
```
```
‣ Type=signal  Endian=l  Flags=1  Version=1 Cookie=318  Timestamp="Fri 2023-09-15 08:05:13.169876 UTC"
  Sender=:1.6  Path=/org/freedesktop/NetworkManager/IP4Config/2  Interface=org.freedesktop.DBus.Properties  Member=PropertiesChanged
  UniqueName=:1.6
  MESSAGE "sa{sv}as" {
          STRING "org.freedesktop.NetworkManager.IP4Config";
          ARRAY "{sv}" {
                  DICT_ENTRY "sv" {
                          STRING "DnsOptions";
                          VARIANT "as" {
                                  ARRAY "s" {
                                  };
                          };
                  };
          };
          ARRAY "s" {
          };
  };

‣ Type=signal  Endian=l  Flags=1  Version=1 Cookie=319  Timestamp="Fri 2023-09-15 08:05:39.535767 UTC"
  Sender=:1.6  Path=/org/freedesktop/NetworkManager/IP4Config/2  Interface=org.freedesktop.DBus.Properties  Member=PropertiesChanged
  UniqueName=:1.6
  MESSAGE "sa{sv}as" {
          STRING "org.freedesktop.NetworkManager.IP4Config";
          ARRAY "{sv}" {
                  DICT_ENTRY "sv" {
                          STRING "DnsOptions";
                          VARIANT "as" {
                                  ARRAY "s" {
                                  };
                          };
                  };
          };
          ARRAY "s" {
          };
  };
```

Same on session-bus:

```
﻿﻿﻿﻿﻿** (cryptomator:2): WARNING **: 10:17:40.123: atk-bridge: get_device_events_reply: unknown signature
```
```
‣ Type=signal  Endian=l  Flags=1  Version=1 Cookie=1612  Timestamp="Fri 2023-09-15 08:17:35.059143 UTC"
  Sender=:1.14  Path=/org/gnome/Shell/Introspect  Interface=org.gnome.Shell.Introspect  Member=RunningApplicationsChanged
  UniqueName=:1.14
  MESSAGE "" {
  };

‣ Type=method_call  Endian=l  Flags=0  Version=1 Cookie=257  Timestamp="Fri 2023-09-15 08:17:35.059473 UTC"
  Sender=:1.64  Destination=:1.14  Path=/org/gnome/Shell/Introspect  Interface=org.gnome.Shell.Introspect  Member=GetRunningApplications
  UniqueName=:1.64
  MESSAGE "" {
  };

‣ Type=method_call  Endian=l  Flags=0  Version=1 Cookie=1613  Timestamp="Fri 2023-09-15 08:17:35.059711 UTC"
  Sender=:1.14  Destination=:1.113  Path=/org/ayatana/NotificationItem/org_cryptomator_Cryptomator/Menu  Interface=com.canonical.dbusmenu  Member=Event
  UniqueName=:1.14
  MESSAGE "isvu" {
          INT32 0;
          STRING "opened";
          VARIANT "i" {
                  INT32 0;
          };
          UINT32 0;
  };

‣ Type=method_call  Endian=l  Flags=0  Version=1 Cookie=1614  Timestamp="Fri 2023-09-15 08:17:35.059743 UTC"
  Sender=:1.14  Destination=:1.113  Path=/org/ayatana/NotificationItem/org_cryptomator_Cryptomator/Menu  Interface=com.canonical.dbusmenu  Member=AboutToShow
  UniqueName=:1.14
  MESSAGE "i" {
          INT32 0;
  };

‣ Type=method_return  Endian=l  Flags=1  Version=1 Cookie=63  ReplyCookie=1613  Timestamp="Fri 2023-09-15 08:17:35.060258 UTC"
  Sender=:1.113  Destination=:1.14
  UniqueName=:1.113
  MESSAGE "" {
  };

‣ Type=method_return  Endian=l  Flags=1  Version=1 Cookie=64  ReplyCookie=1614  Timestamp="Fri 2023-09-15 08:17:35.060283 UTC"
  Sender=:1.113  Destination=:1.14
  UniqueName=:1.113
  MESSAGE "b" {
          BOOLEAN false;
  };

‣ Type=method_return  Endian=l  Flags=1  Version=1 Cookie=1615  ReplyCookie=257  Timestamp="Fri 2023-09-15 08:17:35.063924 UTC"
  Sender=:1.14  Destination=:1.64
  UniqueName=:1.14
  MESSAGE "a{sa{sv}}" {
          ARRAY "{sa{sv}}" {
                  DICT_ENTRY "sa{sv}" {
                          STRING "org.cryptomator.Cryptomator.desktop";
                          ARRAY "{sv}" {
                                  DICT_ENTRY "sv" {
                                          STRING "sandboxed-app-id";
                                          VARIANT "s" {
                                                  STRING "org.cryptomator.Cryptomator";
                                          };
                                  };
                          };
                  };
                  DICT_ENTRY "sa{sv}" {
                          STRING "org.gnome.Console.desktop";
                          ARRAY "{sv}" {
                          };
                  };
          };
  };

‣ Type=method_call  Endian=l  Flags=0  Version=1 Cookie=1616  Timestamp="Fri 2023-09-15 08:17:36.164373 UTC"
  Sender=:1.14  Destination=:1.113  Path=/org/ayatana/NotificationItem/org_cryptomator_Cryptomator/Menu  Interface=com.canonical.dbusmenu  Member=Event
  UniqueName=:1.14
  MESSAGE "isvu" {
          INT32 3;
          STRING "clicked";
          VARIANT "i" {
                  INT32 0;
          };
          UINT32 1290994;
  };

‣ Type=method_return  Endian=l  Flags=1  Version=1 Cookie=65  ReplyCookie=1616  Timestamp="Fri 2023-09-15 08:17:36.165534 UTC"
  Sender=:1.113  Destination=:1.14
  UniqueName=:1.113
  MESSAGE "" {
  };

‣ Type=signal  Endian=l  Flags=1  Version=1 Cookie=1617  Timestamp="Fri 2023-09-15 08:17:36.167500 UTC"
  Sender=:1.14  Path=/org/gnome/Shell/Introspect  Interface=org.gnome.Shell.Introspect  Member=RunningApplicationsChanged
  UniqueName=:1.14
  MESSAGE "" {
  };

‣ Type=method_call  Endian=l  Flags=0  Version=1 Cookie=1618  Timestamp="Fri 2023-09-15 08:17:36.167778 UTC"
  Sender=:1.14  Destination=:1.113  Path=/org/ayatana/NotificationItem/org_cryptomator_Cryptomator/Menu  Interface=com.canonical.dbusmenu  Member=Event
  UniqueName=:1.14
  MESSAGE "isvu" {
          INT32 0;
          STRING "closed";
          VARIANT "i" {
                  INT32 0;
          };
          UINT32 0;
  };

‣ Type=method_call  Endian=l  Flags=0  Version=1 Cookie=258  Timestamp="Fri 2023-09-15 08:17:36.167826 UTC"
  Sender=:1.64  Destination=:1.14  Path=/org/gnome/Shell/Introspect  Interface=org.gnome.Shell.Introspect  Member=GetRunningApplications
  UniqueName=:1.64
  MESSAGE "" {
  };

‣ Type=method_return  Endian=l  Flags=1  Version=1 Cookie=1619  ReplyCookie=258  Timestamp="Fri 2023-09-15 08:17:36.172747 UTC"
  Sender=:1.14  Destination=:1.64
  UniqueName=:1.14
  MESSAGE "a{sa{sv}}" {
          ARRAY "{sa{sv}}" {
                  DICT_ENTRY "sa{sv}" {
                          STRING "org.cryptomator.Cryptomator.desktop";
                          ARRAY "{sv}" {
                                  DICT_ENTRY "sv" {
                                          STRING "active-on-seats";
                                          VARIANT "as" {
                                                  ARRAY "s" {
                                                          STRING "seat0";
                                                  };
                                          };
                                  };
                                  DICT_ENTRY "sv" {
                                          STRING "sandboxed-app-id";
                                          VARIANT "s" {
                                                  STRING "org.cryptomator.Cryptomator";
                                          };
                                  };
                          };
                  };
                  DICT_ENTRY "sa{sv}" {
                          STRING "org.gnome.Console.desktop";
                          ARRAY "{sv}" {
                          };
                  };
          };
  };

‣ Type=method_return  Endian=l  Flags=1  Version=1 Cookie=66  ReplyCookie=1618  Timestamp="Fri 2023-09-15 08:17:36.213698 UTC"
  Sender=:1.113  Destination=:1.14
  UniqueName=:1.113
  MESSAGE "" {
  };

‣ Type=signal  Endian=l  Flags=1  Version=1 Cookie=1620  Timestamp="Fri 2023-09-15 08:17:36.220471 UTC"
  Sender=:1.14  Path=/org/gnome/Shell/Introspect  Interface=org.gnome.Shell.Introspect  Member=WindowsChanged
  UniqueName=:1.14
  MESSAGE "" {
  };

‣ Type=signal  Endian=l  Flags=1  Version=1 Cookie=1621  Timestamp="Fri 2023-09-15 08:17:37.564541 UTC"
  Sender=:1.14  Path=/org/gnome/Shell/Introspect  Interface=org.gnome.Shell.Introspect  Member=WindowsChanged
  UniqueName=:1.14
  MESSAGE "" {
  };

‣ Type=signal  Endian=l  Flags=1  Version=1 Cookie=1622  Timestamp="Fri 2023-09-15 08:17:43.082297 UTC"
  Sender=:1.14  Path=/org/gnome/Shell/Introspect  Interface=org.gnome.Shell.Introspect  Member=RunningApplicationsChanged
  UniqueName=:1.14
  MESSAGE "" {
  };

‣ Type=method_call  Endian=l  Flags=0  Version=1 Cookie=1623  Timestamp="Fri 2023-09-15 08:17:43.082692 UTC"
  Sender=:1.14  Destination=:1.113  Path=/org/ayatana/NotificationItem/org_cryptomator_Cryptomator/Menu  Interface=com.canonical.dbusmenu  Member=Event
  UniqueName=:1.14
  MESSAGE "isvu" {
          INT32 0;
          STRING "opened";
          VARIANT "i" {
                  INT32 0;
          };
          UINT32 0;
  };

‣ Type=method_call  Endian=l  Flags=0  Version=1 Cookie=1624  Timestamp="Fri 2023-09-15 08:17:43.082726 UTC"
  Sender=:1.14  Destination=:1.113  Path=/org/ayatana/NotificationItem/org_cryptomator_Cryptomator/Menu  Interface=com.canonical.dbusmenu  Member=AboutToShow
  UniqueName=:1.14
  MESSAGE "i" {
          INT32 0;
  };

‣ Type=method_call  Endian=l  Flags=0  Version=1 Cookie=259  Timestamp="Fri 2023-09-15 08:17:43.082788 UTC"
  Sender=:1.64  Destination=:1.14  Path=/org/gnome/Shell/Introspect  Interface=org.gnome.Shell.Introspect  Member=GetRunningApplications
  UniqueName=:1.64
  MESSAGE "" {
  };

‣ Type=method_return  Endian=l  Flags=1  Version=1 Cookie=67  ReplyCookie=1623  Timestamp="Fri 2023-09-15 08:17:43.083260 UTC"
  Sender=:1.113  Destination=:1.14
  UniqueName=:1.113
  MESSAGE "" {
  };

‣ Type=method_return  Endian=l  Flags=1  Version=1 Cookie=68  ReplyCookie=1624  Timestamp="Fri 2023-09-15 08:17:43.083285 UTC"
  Sender=:1.113  Destination=:1.14
  UniqueName=:1.113
  MESSAGE "b" {
          BOOLEAN false;
  };

‣ Type=method_return  Endian=l  Flags=1  Version=1 Cookie=1625  ReplyCookie=259  Timestamp="Fri 2023-09-15 08:17:43.087463 UTC"
  Sender=:1.14  Destination=:1.64
  UniqueName=:1.14
  MESSAGE "a{sa{sv}}" {
          ARRAY "{sa{sv}}" {
                  DICT_ENTRY "sa{sv}" {
                          STRING "org.cryptomator.Cryptomator.desktop";
                          ARRAY "{sv}" {
                                  DICT_ENTRY "sv" {
                                          STRING "sandboxed-app-id";
                                          VARIANT "s" {
                                                  STRING "org.cryptomator.Cryptomator";
                                          };
                                  };
                          };
                  };
                  DICT_ENTRY "sa{sv}" {
                          STRING "org.gnome.Console.desktop";
                          ARRAY "{sv}" {
                          };
                  };
          };
  };

‣ Type=method_call  Endian=l  Flags=0  Version=1 Cookie=1626  Timestamp="Fri 2023-09-15 08:17:44.154418 UTC"
  Sender=:1.14  Destination=:1.113  Path=/org/ayatana/NotificationItem/org_cryptomator_Cryptomator/Menu  Interface=com.canonical.dbusmenu  Member=Event
  UniqueName=:1.14
  MESSAGE "isvu" {
          INT32 9;
          STRING "clicked";
          VARIANT "i" {
                  INT32 0;
          };
          UINT32 1298977;
  };

‣ Type=method_return  Endian=l  Flags=1  Version=1 Cookie=69  ReplyCookie=1626  Timestamp="Fri 2023-09-15 08:17:44.155297 UTC"
  Sender=:1.113  Destination=:1.14
  UniqueName=:1.113
  MESSAGE "" {
  };

‣ Type=signal  Endian=l  Flags=1  Version=1 Cookie=1627  Timestamp="Fri 2023-09-15 08:17:44.158725 UTC"
  Sender=:1.14  Path=/org/gnome/Shell/Introspect  Interface=org.gnome.Shell.Introspect  Member=RunningApplicationsChanged
  UniqueName=:1.14
  MESSAGE "" {
  };

‣ Type=method_call  Endian=l  Flags=0  Version=1 Cookie=260  Timestamp="Fri 2023-09-15 08:17:44.159355 UTC"
  Sender=:1.64  Destination=:1.14  Path=/org/gnome/Shell/Introspect  Interface=org.gnome.Shell.Introspect  Member=GetRunningApplications
  UniqueName=:1.64
  MESSAGE "" {
  };

‣ Type=method_call  Endian=l  Flags=0  Version=1 Cookie=1628  Timestamp="Fri 2023-09-15 08:17:44.159381 UTC"
  Sender=:1.14  Destination=:1.113  Path=/org/ayatana/NotificationItem/org_cryptomator_Cryptomator/Menu  Interface=com.canonical.dbusmenu  Member=Event
  UniqueName=:1.14
  MESSAGE "isvu" {
          INT32 0;
          STRING "closed";
          VARIANT "i" {
                  INT32 0;
          };
          UINT32 0;
  };

‣ Type=method_return  Endian=l  Flags=1  Version=1 Cookie=1629  ReplyCookie=260  Timestamp="Fri 2023-09-15 08:17:44.166341 UTC"
  Sender=:1.14  Destination=:1.64
  UniqueName=:1.14
  MESSAGE "a{sa{sv}}" {
          ARRAY "{sa{sv}}" {
                  DICT_ENTRY "sa{sv}" {
                          STRING "org.cryptomator.Cryptomator.desktop";
                          ARRAY "{sv}" {
                                  DICT_ENTRY "sv" {
                                          STRING "active-on-seats";
                                          VARIANT "as" {
                                                  ARRAY "s" {
                                                          STRING "seat0";
                                                  };
                                          };
                                  };
                                  DICT_ENTRY "sv" {
                                          STRING "sandboxed-app-id";
                                          VARIANT "s" {
                                                  STRING "org.cryptomator.Cryptomator";
                                          };
                                  };
                          };
                  };
                  DICT_ENTRY "sa{sv}" {
                          STRING "org.gnome.Console.desktop";
                          ARRAY "{sv}" {
                          };
                  };
          };
  };

‣ Type=signal  Endian=l  Flags=1  Version=1 Cookie=54  Timestamp="Fri 2023-09-15 08:17:44.169392 UTC"
  Sender=org.freedesktop.DBus  Destination=:1.115  Path=/org/freedesktop/DBus  Interface=org.freedesktop.DBus  Member=NameLost
  MESSAGE "s" {
          STRING ":1.115";
  };

‣ Type=signal  Endian=l  Flags=1  Version=1 Cookie=100  Timestamp="Fri 2023-09-15 08:17:44.169434 UTC"
  Sender=org.freedesktop.DBus  Path=/org/freedesktop/DBus  Interface=org.freedesktop.DBus  Member=NameOwnerChanged
  MESSAGE "sss" {
          STRING ":1.115";
          STRING ":1.115";
          STRING "";
  };

‣ Type=signal  Endian=l  Flags=1  Version=1 Cookie=51  Timestamp="Fri 2023-09-15 08:17:44.170323 UTC"
  Sender=org.freedesktop.DBus  Destination=:1.114  Path=/org/freedesktop/DBus  Interface=org.freedesktop.DBus  Member=NameLost
  MESSAGE "s" {
          STRING ":1.114";
  };
```

**Comment by purejava** [comment]
Fixed with appindicator-gtk3-java release `1.3.4`.
